thespian/system/transport/TCPTransport.py

```diff
@@ -344,9 +344,9 @@ def getAddressFromString(addrspec, adminRouting=False):
         else:
             addrparts = addrspec.split(':')
         addrtype = RoutedTCPv4ActorAddress if adminRouting else TCPv4ActorAddress
-        if 1 == len(addrparts):
-            return ActorAddress(addrtype(addrparts[0], DEFAULT_ADMIN_PORT, external=True))
-        return ActorAddress(addrtype(addrparts[0], addrparts[1], external=True))
+        return ActorAddress(addrtype(addrparts[0],
+                                     addrparts[1] if addrparts[1:] else DEFAULT_ADMIN_PORT,
+                                     external=True))
```

Comment:
I think this is the best of the options we discussed: no temp variables, no mutation, but still clear and concise.